geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloQueryPlan.scala

```diff
@@ -124,23 +135,29 @@ case class BatchScanPlan(filter: AccumuloFilterStrategyType,
     scanEntries(ds).map(entriesToFeatures)
 
   def scanEntries(ds: AccumuloDataStore, auths: Option[Authorizations] = None): CloseableIterator[Entry[Key, Value]] = {
-    if (ranges.isEmpty) { CloseableIterator.empty } else {
-      val batchRanges = AccumuloQueryProperties.SCAN_BATCH_RANGES.option.map(_.toInt).getOrElse(Int.MaxValue)
-      val batched = ranges.grouped(batchRanges)
-      SelfClosingIterator(batched).flatMap { ranges =>
-        val scanner = ds.connector.createBatchScanner(table, auths.getOrElse(ds.auths), numThreads)
-        scanner.setRanges(ranges)
-        configure(scanner)
-        SelfClosingIterator(scanner.iterator, scanner.close())
-      }
+    if (PartitionParallelScan.toBoolean.contains(true)) {
+      // kick off all the scans at once
+      tables.map(scanner(ds, auths, _)).foldLeft(CloseableIterator.empty[Entry[Key, Value]])(_ ++ _)
+    } else {
+      // kick off the scans sequentially as they finish
+      SelfClosingIterator(tables.iterator).flatMap(scanner(ds, auths, _))
```

Comment:
In either case, does the number of query threads/resources get applied?

And in the sequential case, does one see a 'gating' effect?  I.e. 10 query threads happen against one table until they all finish and then the threads move onto the next table?

Reply:
each 'query' (against at different table) will use the normal number of threads, so running in parallel will multiply the total threads. in the sequential case, one 'query' won't start until the previous one finishes.

Reply:
Ok, so pertaining to another question... flattening the list of queries out to use one common pool would be another execution option and would require additional development.

Sounds like a new ticket.:)

Reply:
Well the thread pool is currently managed by the batch scanner, so I'm not actually sure that it would be possible to share a thread pool. You could divide the number of threads among the various scans, but that doesn't seem like a great solution either.

Reply:
Sounds great.  I thought we may have been managing the thread.  Since we are not, this is an excellent way to handle this.

Reply:
we do handle the threading in hbase, so if you think it's worthwhile we could do something there. It may be possible to hook into the accumulo API too, but not that I know of.

Reply:
Eh, how about we toss a ticket in for managing one thread pool in HBase?  (I.e., let's think about it for a few days...)

The Accumulo case is subtle.  @jahhulbert-ccri do you know if there's way to share a thread pool among BatchScanners?!